noir-projects/noir-contracts/contracts/uniswap_contract/src/main.nr

```diff
@@ -124,11 +123,17 @@ pub contract Uniswap {
         secret_hash_for_L1_to_l2_message: Field, // for when l1 uniswap portal inserts the message to consume output assets on L2
         caller_on_L1: EthAddress, // ethereum address that can call this function on the L1 portal (0x0 if anyone can call)
     ) {
+        let input_asset_bridge_config =
+            TokenBridge::at(input_asset_bridge).get_config().view(&mut context);
+        let output_asset_bridge_config =
+            TokenBridge::at(output_asset_bridge).get_config().view(&mut context);
+
         // Assert that user provided token address is same as expected by token bridge.
         // we can't directly use `input_asset_bridge.token` because that is a public method and public can't return data to private
-        Uniswap::at(context.this_address())
-            ._assert_token_is_same(input_asset, input_asset_bridge)
-            .enqueue_view(&mut context);
+        assert(
+            input_asset.eq(input_asset_bridge_config.token),
+            "input_asset address is not the same as seen in the bridge contract",
+        );
```

Comment:
Same as in token bridge I think this was done before we were able to read public storage from private. I just replaced that with the inlined assert above.